middleware/proxy/upstream.go

```diff
@@ -159,10 +159,16 @@ func parseBlock(c *parse.Dispenser, u *staticUpstream) error {
 		if !c.Args(&header, &value) {
 			return c.ArgErr()
 		}
-		proxyHeaders.Add(header, value)
+		if u.proxyHeaders == nil {
+			u.proxyHeaders = make(http.Header)
+		}
+		u.proxyHeaders.Add(header, value)
 	case "websocket":
-		proxyHeaders.Add("Connection", "{>Connection}")
-		proxyHeaders.Add("Upgrade", "{>Upgrade}")
+		if u.proxyHeaders == nil {
+			u.proxyHeaders = make(http.Header)
+		}
```

Comment:
Instead of this nil check (and the one above on line 162), what if we always set `u.proxyHeaders` to `make(http.Header)` when the staticUpstream is created, way up by line 42?